### DISCUSSION TITLE
Server-side rendering (SSR) for SEO, faster loading and social sharing

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/nicegui#663] [Ideas / Feature Requests]
As written in https://github.com/zauberzeug/nicegui/discussions/637#discussioncomment-5447487, SSR would especially beneficial for SEO, faster loading and social sharing. Because #637 is focusing on static-site generation (SSR), this ticket will focus on SSR as suggested by @falkoschindler in https://github.com/zauberzeug/nicegui/discussions/637#discussioncomment-5448349.

**Comment by falkoschindler** [comment]
It looks like SEO isn't as bad as expected. Google Page Speed rates SEO for nicegui.io with 100%:
https://pagespeed.web.dev/analysis/https-nicegui-io/2xzi6ku0ky
Of course, there are other deficits, which we are working on (#658). But maybe SEO is not an issue.

**Comment by evnchn** [comment]
A tough one: progress at https://github.com/evnchn/nicegui/pull/83
